node/src/blockchain/test_utils.rs

```diff
@@ -321,176 +318,6 @@ pub fn make_fake_event_loop_handle() -> EventLoopHandle {
         .0
 }
 
-#[derive(Default)]
-pub struct BatchPayableToolsFactoryMock<T> {
-    make_results: RefCell<Vec<Box<dyn BatchPayableTools<T>>>>,
-}
-
-impl<T> BatchPayableToolsFactoryMock<T> {
-    pub fn make_result(self, result: Box<dyn BatchPayableTools<T>>) -> Self {
-        self.make_results.borrow_mut().push(result);
-        self
-    }
-}
-
-#[derive(Default)]
-pub struct BatchPayableToolsMock<T: BatchTransport> {
```

Comment:
A better thing to do would be to make changes in-place rather than moving it into a different file. It would make it easier to see the changes in the diff. Although, I don't want you to do it now, but I wish if you do it in the future PRs.

Reply:
I understand your point, but I did do it as considerately as possible. It looks messy in the review but the resulting state is what matters here.  I had to create a fully new hierarchy in the module blockchain because this area has advanced to a really multi-functional module with a lot of distinct parts, I felt it was urgent to make it maintainable within well defined separate files. It's really hard to do this right, and I did try hard, so please don't let you have quick prejudices. :pray:

Reply:
It's alright @bertllll! It wasn't too hard for me to verify it. I'm glad to know that we're on the same page on this. 🙏